### PR TITLE
refactor(compiler): use new ngc for i18n

### PR DESCRIPTION
**Pull request by tbosch**
This also changes ngc to support all tsc command line arguments.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Comment by tbosch** [comment]
The commit is a variation from https://github.com/angular/angular/pull/18966/commits/5fcad62ddd66764ca5f15ea52ce9e3f7c03b0f42 which @matsko and I authored together.

File: packages/compiler-cli/src/codegen.ts [hunk 1]
```diff
@@ -91,17 +98,17 @@ export class CodeGenerator {
...
       missingTranslation = compiler.core.MissingTranslationStrategy.Ignore;
     }
     const {compiler: aotCompiler} = compiler.createAotCompiler(ngCompilerHost, {
       translations: transContent,
-      i18nFormat: cliOptions.i18nFormat || undefined,
-      locale: cliOptions.locale || undefined, missingTranslation,
-      enableLegacyTemplate: options.enableLegacyTemplate === true,
+      i18nFormat: i18nOptions.i18nFormat || undefined,
+      locale: i18nOptions.locale || undefined, missingTranslation,
+      enableLegacyTemplate: options.enableLegacyTemplate !== false,
```
Comment (vicb) [inline packages/compiler-cli/src/codegen.ts]: is `enableLegacyTemplate` intended ?

File: packages/compiler-cli/src/extract_i18n.ts [hunk 2]
```diff
@@ -13,29 +13,38 @@
...
+}
 
-import {Extractor} from './extractor';
+function readXi18nCommandLineAndConfiguration(args: string[]): ParsedConfiguration {
+  const options: api.CompilerOptions = {};
+  const parsedArgs = require('minimist')(args);
+  if (parsedArgs.outFile) options.i18nOutFile = parsedArgs.outFile;
+  if (parsedArgs.i18nFormat) options.i18nOutFormat = parsedArgs.i18nFormat;
+  if (parsedArgs.locale) options.i18nOutLocale = parsedArgs.locale;
+  const mt = parsedArgs.missingTranslation;
```
Comment (vicb) [inline packages/compiler-cli/src/extract_i18n.ts]: missing translation is a merge thing only

File: packages/compiler-cli/src/codegen.ts (hunk 1 above)
Comment (tbosch) [reply packages/compiler-cli/src/codegen.ts]: Good catch! The commit was old and I missed this when rebasing...

File: packages/compiler-cli/src/extract_i18n.ts (hunk 2 above)
Comment (tbosch) [reply packages/compiler-cli/src/extract_i18n.ts]: Removed.

**Review by vicb — APPROVED** [review]

**Pull request closed without merging**